packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -188,8 +193,10 @@ class FlutterCommandRunner extends CommandRunner<void> {
   Future<void> runCommand(ArgResults topLevelResults) async {
     final Map<Type, Object?> contextOverrides = <Type, Object?>{};
 
-    // If the disable-telemetry flag has been passed, return out
-    if (topLevelResults.wasParsed('disable-telemetry')) {
+    // If the flag for enabling or disabling telemetry is passed in,
+    // we will return out
+    if (topLevelResults.wasParsed('disable-telemetry') ||
```

Comment:
You should also check and make sure that both aren't specified at the same time, and error out if they are. You'll want a test for that too.

Reply:
Great call, included in the latest commit